src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -166,6 +164,37 @@ public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> ge
             }
         }
 
+        private static TraceId ParseTraceId<T>(T headers, string headerName)
+            where T : IHeadersCollection
+        {
+            var headerValues = headers.GetValues(headerName);
+            return ParseTraceId(headerValues, headerName);
+        }
+
+        private static TraceId ParseTraceId<T>(T carrier, Func<T, string, IEnumerable<string>> getter, string headerName)
+        {
+            var headerValues = getter(carrier, headerName).ToList();
```

Comment:
I think the `ToList()` conversions are unnecessary since we're only enumerating the items

Reply:
Removed